### ISSUE TITLE
Error when loading /nbextensions: "Expecting property name enclosed in double quotes"

### ISSUE THREAD

**Issue by masnick** [ipython-contrib/IPython-notebook-extensions#555]
Hi - I just installed notebook extensions with `pip install` (so I'm using ddb5810 on master). I'm running Python 2.7.10 with `pyenv` and Jupyter 4.1.0.

When I try to go to `http://localhost:8888/nbextensions`, I get the following error/stack trace:

```
[I 20:32:08.405 NotebookApp] Found 4.x extension SKILL Syntax
[I 20:32:08.406 NotebookApp] Found 4.x extension NbExtensions menu item
[I 20:32:08.408 NotebookApp] Found 3.x, 4.x extension Gist-it
[I 20:32:08.411 NotebookApp] Found 4.x extension Printview
[I 20:32:08.412 NotebookApp] Found 4.x extension table_beautifier
[I 20:32:08.415 NotebookApp] Found 4.x extension zenmode
[I 20:32:08.419 NotebookApp] Found 3.x, 4.x extension AutoSaveTime
[I 20:32:08.423 NotebookApp] Found 3.x, 4.x extension Autoscroll
[I 20:32:08.424 NotebookApp] Found 4.x extension Chrome Clipboard
[I 20:32:08.425 NotebookApp] Found 4.x extension Code Font Size
[I 20:32:08.427 NotebookApp] Found 4.x extension Codefolding
[I 20:32:08.439 NotebookApp] Found 4.x extension Collapsible Headings
[I 20:32:08.449 NotebookApp] Found 3.x, 4.x extension Comment/Uncomment Hotkey
[I 20:32:08.451 NotebookApp] Found 4.x extension datestamper
[I 20:32:08.453 NotebookApp] Found 4.x extension Drag and Drop
[I 20:32:08.455 NotebookApp] Found 4.x extension Equation Auto Numbering
[I 20:32:08.457 NotebookApp] Found 4.x extension ExecuteTime
[I 20:32:08.459 NotebookApp] Found 4.x extension Exercise
[I 20:32:08.460 NotebookApp] Found 4.x extension Exercise2
[I 20:32:08.462 NotebookApp] Found 4.x extension Freeze
[I 20:32:08.464 NotebookApp] Found 3.x, 4.x extension Help panel
[I 20:32:08.465 NotebookApp] Found 3.x 4.x extension Hide input
[I 20:32:08.467 NotebookApp] Found 3.x 4.x extension Hide input all
[I 20:32:08.468 NotebookApp] Found 3.x, Jupyter (4.x) extension highlighter
[I 20:32:08.469 NotebookApp] Found 3.x, 4.x extension Initialization cells
[I 20:32:08.475 NotebookApp] Found 4.x extension Keyboard shortcut editor
[I 20:32:08.476 NotebookApp] Found 3.x, Jupyter (4.x) extension (some) LaTeX environments for Jupyter
[I 20:32:08.479 NotebookApp] Found 4.x extension Limit Output
[I 20:32:08.480 NotebookApp] Found 4.x extension Move selected cells
[I 20:32:08.482 NotebookApp] Found 4.x extension Navigation-Hotkeys
[I 20:32:08.484 NotebookApp] Found 4.x extension Notify
[I 20:32:08.486 NotebookApp] Found 4.x extension Python Markdown
[I 20:32:08.487 NotebookApp] Found 4.x extension Launch QTConsole
[I 20:32:08.488 NotebookApp] Found 4.x extension Rubberband
[I 20:32:08.499 NotebookApp] Found 4.x extension Ruler
[I 20:32:08.501 NotebookApp] Found 4.x extension Runtools
[I 20:32:08.503 NotebookApp] Found 4.x extension Scratchpad
[I 20:32:08.505 NotebookApp] Found 4.x extension Search-Replace
[I 20:32:08.506 NotebookApp] Found 4.x extension Skip-Traceback
[I 20:32:08.511 NotebookApp] Found 4.x extension spellchecker
[I 20:32:08.523 NotebookApp] Found 4.x extension Table of Contents (2)
[I 20:32:08.526 NotebookApp] Found 3.x 4.x extension Toggle all line numbers
[I 20:32:08.527 NotebookApp] Found 4.x extension Tree Filter
[E 20:32:08.801 NotebookApp] Unhandled error in API request
    Traceback (most recent call last):
      File "/Users/max/.pyenv/versions/data-2.7.10/lib/python2.7/site-packages/notebook/base/handlers.py", line 458, in wrapper
        result = yield gen.maybe_future(method(self, *args, **kwargs))
      File "/Users/max/.pyenv/versions/data-2.7.10/lib/python2.7/site-packages/notebook/services/config/handlers.py", line 20, in get
        self.finish(json.dumps(self.config_manager.get(section_name)))
      File "/Users/max/.pyenv/versions/data-2.7.10/lib/python2.7/site-packages/traitlets/config/manager.py", line 63, in get
        return json.load(f)
      File "/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/__init__.py", line 290, in load
        **kw)
      File "/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/__init__.py", line 338, in loads
        return _default_decoder.decode(s)
      File "/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/decoder.py", line 382, in raw_decode
        obj, end = self.scan_once(s, idx)
    ValueError: Expecting property name enclosed in double quotes: line 12 column 1 (char 152)
[E 20:32:08.802 NotebookApp] {
      "Accept-Language": "en-US,en;q=0.8", 
      "Accept-Encoding": "gzip, deflate, sdch", 
      "X-Requested-With": "XMLHttpRequest", 
      "Host": "localhost:8888", 
      "Accept": "application/json, text/javascript, */*; q=0.01", 
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.87 Safari/537.36", 
      "Connection": "keep-alive", 
      "Referer": "http://localhost:8888/nbextensions", 
      "Cookie": "email=xxxx; csrftoken=xxxx; hid=1"
    }
[E 20:32:08.803 NotebookApp] 500 GET /api/config/notebook?_=1458873128677 (::1) 1.50ms referer=http://localhost:8888/nbextensions
```

**Comment by juhasch** [comment]
Can you post what `http://localhost:8888/api/config/notebook` returns in your Browser ?

**Comment by jcb91** [comment]
This looks like the config API attempting to read a json config file which isn't properly formatted json - most likely there's a trailing comma left in (which json doesn't allow)

**Comment by masnick** [comment]
@jcb91 I agree, but it doesn't appear to say wh

@juhasch here's what I get at `http://localhost:8888/api/config/notebook`:

```
{"reason": null, "traceback": "Traceback (most recent call last):\n  File \"/Users/max/.pyenv/versions/data-2.7.10/lib/python2.7/site-packages/notebook/base/handlers.py\", line 458, in wrapper\n    result = yield gen.maybe_future(method(self, *args, **kwargs))\n  File \"/Users/max/.pyenv/versions/data-2.7.10/lib/python2.7/site-packages/notebook/services/config/handlers.py\", line 20, in get\n    self.finish(json.dumps(self.config_manager.get(section_name)))\n  File \"/Users/max/.pyenv/versions/data-2.7.10/lib/python2.7/site-packages/traitlets/config/manager.py\", line 63, in get\n    return json.load(f)\n  File \"/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/__init__.py\", line 290, in load\n    **kw)\n  File \"/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/__init__.py\", line 338, in loads\n    return _default_decoder.decode(s)\n  File \"/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/decoder.py\", line 366, in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n  File \"/Users/max/.pyenv/versions/2.7.10/lib/python2.7/json/decoder.py\", line 382, in raw_decode\n    obj, end = self.scan_once(s, idx)\nValueError: Expecting property name enclosed in double quotes: line 12 column 1 (char 152)\n", "message": "Unknown server error"}
```

**Comment by jcb91** [comment]
@masnick Ah, I see, apologies, I should have twigged this wasn't obvious.

Well, the nbextensions config extension loads four config files: `notebook.json`, `tree.json`, `edit.json`, and `common.json`.

These config files are per-user files, so should exist in the `nbconfig` subdirectory of `jupyter_config_dir()` as reported by `jupyter_core.paths`. From the fact that you get an error at `http://localhost:8888/api/config/notebook`, it seems that at least your notebook.json has an error in, although of course there may be others in the other files. You can use the following snippet to locate the exisiting config files:

```python
import os.path
from notebook.services.config import ConfigManager
cm = ConfigManager()
for section in ('edit', 'tree', 'notebook', 'common'):
    conf_path = cm.file_name(section)
    if os.path.exists(conf_path):
        print(conf_path)
```

perhaps I ought to add a try/catch clause to the javascript, and provide a better error message...

**Comment by jcb91** [comment]
(although, tbh, I think the config API should probably tell you what file it's failed to load in its own log message!)

**Comment by jcb91** [comment]
closing as this is covered by https://github.com/ipython/traitlets/issues/203

**Comment by cprior** [comment]
Was also bitten by this after an attempt to install my first nbextension ever (rise live slides).

Not 100% sure how I solved it, but it was one of:
- having permission problems in /home/john/.jupyter
- an emtpy file notebook.json (maybe a stupid attempt to disable said extension)

I found the above mentioned files
`notebook.json`, `tree.json`, `edit.json`, and `common.json`
in either
```
/root/.jupyter/nbconfig
/home/john/.jupyter/nbconfig
/usr/local/etc/jupyter/nbconfig
/usr/etc/jupyter/nbconfig
```
Might help others (this issue was the "googliest") to troubleshoot.

From what I understand in https://github.com/ipython/traitlets/issues/203 future versions of notebook will spit out the offending json file.
